sentence_transformers/sparse_encoder/models/IDF.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+import json
+import os
+from typing import TYPE_CHECKING, Any
+
+import torch
+from safetensors.torch import load_file as load_safetensors_file
+from safetensors.torch import save_model as save_safetensors_model
+from transformers import AutoTokenizer
+
+if TYPE_CHECKING:
+    from transformers import PreTrainedTokenizer
+
+
+class IDF(torch.nn.Module):
+    def __init__(
+        self,
+        weight: torch.Tensor,
+        frozen: bool = False,
+        tokenizer: PreTrainedTokenizer | None = None,
```

Comment:
```suggestion
        tokenizer: PreTrainedTokenizer,
        frozen: bool = False,
```
This makes it fully required